clients/client/src/main/java/org/projectnessie/client/NessieConfigConstants.java

```diff
@@ -28,13 +28,17 @@
   /**
    * Config property name ({@value #CONF_NESSIE_USERNAME}) for the user name used for (basic)
    * authentication.
+   *
+   * @deprecated "basic" HTTP authentication is not considered secure.
    */
-  public static final String CONF_NESSIE_USERNAME = "nessie.username";
+  @Deprecated public static final String CONF_NESSIE_USERNAME = "nessie.authentication.username";
```

Comment:
So to understand this correctly, the legacy is `nessie.username` and the new one `nessie.authentication.username`? 
If yes,  would it make sense to just keep the legacy config variables but only with `@Deprecated` annotation? E.g:
```
@Deprecated public static final String CONF_NESSIE_USERNAME = "nessie.username";
public static final String CONF_NESSIE_AUTHENTICATION_USERNAME = "nessie.authentication.username"
```
And then we can use the legacy config variables to fallback as you already doing [there](https://github.com/projectnessie/nessie/pull/1968/files#diff-16b70889c228cd93cc9caa0811e507ffe3f350b3b420790f2cc02be2ae2859b1R51) but using the deprecated variable.

Reply:
`@Deprecated` is also about the security implications, plus production Nessie server do not support username/password (aka "basic") authentication.

Reply:
ok makes sense